src/redux/features/tracking/sitePopulationSelector.ts

```diff
@@ -1,5 +1,26 @@
 import { RootState } from 'src/redux/rootReducer';
+import { createCachedSelector } from 're-reselect';
 
 export const selectSitePopulation = (state: RootState) => state.sitePopulation?.zones;
 
 export const selectSitePopulationError = (state: RootState) => state.sitePopulation?.error;
+
+export const selectZonePopulationStats = createCachedSelector(
+  (state: RootState) => selectSitePopulation(state),
+  (zones) => {
+    const zoneStats: Record<number, { name: string; reportedPlants: number; reportedSpecies: number }> = {};
+    zones?.forEach((zone) => {
+      const speciesSet = new Set<string>();
+      let numPlants = 0;
+      zone.plantingSubzones?.forEach((subzone) =>
+        subzone.populations?.forEach((population) => {
+          const count = +population['totalPlants(raw)'];
+          numPlants += isNaN(count) ? 0 : count;
+          speciesSet.add(population.species_scientificName);
+        })
+      );
+      zoneStats[+zone.id] = { name: zone.name, reportedPlants: numPlants, reportedSpecies: speciesSet.size };
+    });
+    return zoneStats;
+  }
+)(() => 'zonePopulationStats');
```

Comment:
for this use-case we can use `createSelector` which maintains a cache of 1 entry.
CachedSelector was introduced to support multiple cache entries (helpful when the cache key is dynamic).